BoxContentSDK/BoxContentSDK/Operations/BOXAPIOperation.m

```diff
@@ -540,18 +540,26 @@ - (void)processResponse:(NSURLResponse *)response
 
 - (void)finishURLSessionTaskWithData:(NSData *)data response:(NSURLResponse *)response error:(NSError *)error
 {
-    if (self.HTTPResponse == nil) {
+    if (self.HTTPResponse == nil && response != nil) {
         // If the response has not already been handled (some, but not all, operations get
-        // 'processResponse:' earlier in the process) handle it here.
+        // 'processResponse:' earlier in the process), handle it here.
+        // Also, do not handle a nil response. That could happen if there was a failure on the whole
+        // network connection, for example. In that case, there is no response.
```

Comment:
can we add comment for processResponse in BOXAPIOperation.h/.m that this method is expecting a valid response for processing, else that is considered an error.

Separately, should we check if self.error is nil inside processResponse before overriding it?